liquidation-monitor/src/agent.js

```diff
@@ -0,0 +1,383 @@
+const config = require('../agent-config.json');
+const {
+  getAbi, callCompoundAPI, buildJsonRequest,
+} = require('./utils');
+
+// Stores information about each account
+const initializeData = {};
+
+// Global functions
+function createAlert(
+  developerAbbreviation,
+  protocolName,
+  protocolAbbreviation,
+  type,
+  severity,
+  borrowerAddress,
+  liquidationAmount,
+  shortfallAmount,
+  healthFactor,
+) {
+  return Finding.fromObject({
+    name: `${protocolName} Liquidation Threshold Alert`,
+    description: `The address ${borrowerAddress} has dropped below the liquidation threshold. `
+      + `The account may be liquidated for: $${liquidationAmount} USD`,
+    alertId: `${developerAbbreviation}-${protocolAbbreviation}-LIQUIDATION-THRESHOLD`,
+    type: FindingType[type],
+    severity: FindingSeverity[severity],
+    protocol: protocolName,
+    metadata: {
+      borrowerAddress,
+      liquidationAmount,
+      shortfallAmount,
+      healthFactor,
+    },
+  });
+}
+
+async function verifyToken(data, tokenAddressImport) {
+  const { tokens } = data;
+  const tokenAddress = tokenAddressImport.toLowerCase();
+  if (tokens[tokenAddress] === undefined) {
+    tokens[tokenAddress] = {};
+    tokens[tokenAddress].contract = new ethers.Contract(
+      tokenAddress, data.cTokenABI, data.provider,
+    );
+    const cContract = tokens[tokenAddress].contract;
+    tokens[tokenAddress].symbol = await cContract.symbol();
+
+    // cETH does not have an underlying contract, so link it to wETH instead
+    if (tokenAddress === '0x4ddc2d193948926d02f9b1fe9e1daa0718270ed5') {
+      tokens[tokenAddress].underlying = '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2';
+    } else {
+      tokens[tokenAddress].underlying = await cContract.underlying();
+    }
+    const exchangeRate = await cContract.exchangeRateStored();
+    tokens[tokenAddress].cTokenDecimals = BigNumber(
+      (await cContract.decimals()).toString(),
+    );
+
+    // Look up decimals of the underlying tokens as well.
+    const decimalsABI = '["function decimals() view returns (uint)"]';
+    const underlyingTokenContract = new ethers.Contract(
+      tokens[tokenAddress].underlying, decimalsABI, data.provider,
+    );
+    tokens[tokenAddress].tokenDecimals = BigNumber(
+      (await underlyingTokenContract.decimals()).toString(),
+    );
+
+    tokens[tokenAddress].tokenDecimalsMult = BigNumber(10).pow(tokens[tokenAddress].tokenDecimals);
+    tokens[tokenAddress].cTokenDecimalsMult = BigNumber(10)
+      .pow(tokens[tokenAddress].cTokenDecimals);
+    tokens[tokenAddress].exchangeRate = BigNumber(exchangeRate.toString());
+
+    // Adjusting the multiplier for easier use later.
+    tokens[tokenAddress].exchangeRateMult = BigNumber(exchangeRate.toString())
+      .dividedBy(tokens[tokenAddress].tokenDecimalsMult)
+      .dividedBy(tokens[tokenAddress].cTokenDecimalsMult)
+      .dividedBy(100); // Not sure where this 100 comes from I didn't see it in the docs
```

Comment:
It looks like the value returned from the `exchangeRateStored` method (and the internal `exchangeRateStoredInternal` method) is scaled by 1e18:
https://github.com/compound-finance/compound-protocol/blob/3affca87636eecd901eb43f81a4813186393905d/contracts/CToken.sol#L326
https://github.com/compound-finance/compound-protocol/blob/3affca87636eecd901eb43f81a4813186393905d/contracts/CToken.sol#L337


Note: Be certain not to use ethers.js BigNumber, as the division will cause a loss of precision in the result.  Only use bignumber.js BigNumber, as it will retain precision through the division.
```suggestion
const exchangeRateDecimals = 18;
const multiplier = new BigNumber(10).pow(exchangeRateDecimals);
tokens[tokenAddress].exchangeRateMult = BigNumber(exchangeRate.toString()).div(multiplier);
```

Reply:
“The current exchange rate as an unsigned integer, scaled by 1 * 10^(18 - 8 + Underlying Token Decimals)” - https://compound.finance/docs/ctokens#exchange-rate

The 8 mentioned here is the cToken Decimals. The exchange rate is not a static 1e18. Example, these are the exchange rates for a few:
225975924421044 for USDC ( uses 6 decimals)
20204487858136665 for wBTC ( uses 8 decimals)
202562285509376818212946300 for UNI ( uses 18 decimals )

So the exchange rate / cToken decimals / token decimals yields the actual rate I need.
Given: 1 cUSDC = 0.02259 USDC
For USDC: 225975924421044 / 10^8 / 10^6  = 2.259  This is still off by 100, so I divided by 100 to get the final multiplier of 0.02259 . 

Also “When a market is launched, the cToken exchange rate (how much ETH one cETH is worth) begins at 0.020000 — and increases at a rate equal to the compounding market interest rate.” - https://compound.finance/docs/ctokens#introduction

This might explain the 100 factor to get exchange rate of 2 to 0.02

Reply:
Actually, this part explains it the best. "1 * 10^(18 - 8 + Underlying Token Decimals)"
10^(10 + Underlying Token Decimals). I'll adjust my code to this.

Reply:
Simplified and added more notes in commit 12d462d1b7a29c7db85c9619d9875740f6a01f13